components/ipfs/ipfs_service_browsertest.cc

```diff
@@ -266,60 +312,163 @@ IN_PROC_BROWSER_TEST_F(IpfsServiceBrowserTest, GetAddressesConfigServerError) {
+                       CannotFetchIPFSResourcesFromHTTP) {
   ResetTestServer(
       base::BindRepeating(&IpfsServiceBrowserTest::HandleEmbeddedSrvrRequest,
                           base::Unretained(this)));
-  SetIPFSDefaultGatewayForTest(GetURL("/"));
-  ui_test_utils::NavigateToURL(browser(), GetURL("/simple.html"));
+  SetIPFSDefaultGatewayForTest(GetURL("a.com", "/"));
+  ui_test_utils::NavigateToURL(browser(), GetURL("b.com", "/simple.html"));
   bool as_expected = false;
   content::WebContents* contents =
       browser()->tab_strip_model()->GetActiveWebContents();
   ASSERT_TRUE(ExecuteScriptAndExtractBool(
       contents,
       "fetch('ipfs://"
-      "bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq')"
+      "Qmc2JTQo4iXf24g98otZmGFQq176eQ2Cdbb88qA5ToMEvC/2')"
+      "  .catch((e) => {"
+      "        window.domAutomationController.send(true);"
+      "  });",
+      &as_expected));
+  ASSERT_TRUE(as_expected);
+}
+
+// Make sure an window.fetch works within the ipfs:// scheme
+IN_PROC_BROWSER_TEST_F(IpfsServiceBrowserTest, CanFetchIPFSResourcesFromIPFS) {
+  ResetTestServer(
+      base::BindRepeating(&IpfsServiceBrowserTest::HandleEmbeddedSrvrRequest,
+                          base::Unretained(this)));
+  SetIPFSDefaultGatewayForTest(GetURL("dweb.link", "/"));
+  GURL url("ipfs://Qmc2JTQo4iXf24g98otZmGFQq176eQ2Cdbb88qA5ToMEvC");
+  ui_test_utils::NavigateToURL(browser(), url);
+  bool as_expected = false;
+  content::WebContents* contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+  ASSERT_TRUE(ExecuteScriptAndExtractBool(
+      contents,
+      "fetch('ipfs://"
+      "Qmc2JTQo4iXf24g98otZmGFQq176eQ2Cdbb88qA5ToMEvC/2')"
       "  .then(response => { response.text()"
       "      .then((response_text) => {"
-      "        const result = response_text == 'test content 1';"
+      "        const result = response_text == 'simple content 2';"
       "        window.domAutomationController.send(result);"
       "      })})"
       ".catch((x) => console.log('error: ' + x));",
       &as_expected));
   ASSERT_TRUE(as_expected);
 }
 
-IN_PROC_BROWSER_TEST_F(IpfsServiceBrowserTest, ExternalHandlerMainFrameOnly) {
+// Make sure an <iframe src="ipfs://..."> cannot load within http:// scheme
+IN_PROC_BROWSER_TEST_F(IpfsServiceBrowserTest, CannotLoadIframeFromHTTP) {
   ResetTestServer(
       base::BindRepeating(&IpfsServiceBrowserTest::HandleEmbeddedSrvrRequest,
                           base::Unretained(this)));
-  SetIPFSDefaultGatewayForTest(GetURL("/"));
-  ui_test_utils::NavigateToURL(browser(), GetURL("/simple.html"));
-  bool as_expected = false;
+  SetIPFSDefaultGatewayForTest(GetURL("b.com", "/"));
+  ui_test_utils::NavigateToURL(browser(), GetURL("b.com", "/iframe.html"));
   content::WebContents* contents =
       browser()->tab_strip_model()->GetActiveWebContents();
-  ASSERT_TRUE(ExecuteScriptAndExtractBool(
+
+  std::string location;
+  ASSERT_TRUE(ExecuteScriptAndExtractString(
```

Comment:
`ExecuteScriptAndExtractString` is deprecated in favor of `EvalJs` (see `browser_test_utils.h`). Also it should be possible to set an onload handler (i.e. avoid using the timer)

Reply:
ok thanks